usersession/agent/rest_api.go

```diff
@@ -46,3 +59,129 @@ func sessionInfo(c *Command, r *http.Request) Response {
 	}
 	return SyncResponse(m)
 }
+
+type serviceInstruction struct {
+	Action   string   `json:"action"`
+	Services []string `json:"services"`
+}
+
+var (
+	stopTimeout = time.Duration(timeout.DefaultTimeout)
+	killWait    = 5 * time.Second
+)
+
+func stopOneService(service string, sysd systemd.Systemd) error {
+	err := sysd.Stop(service, stopTimeout)
+	if err != nil && systemd.IsTimeout(err) {
+		// ignore errors for kill; nothing we'd do differently at this point
+		sysd.Kill(service, "TERM", "")
+		time.Sleep(killWait)
+		sysd.Kill(service, "KILL", "")
+	}
+	return err
+}
+
+func serviceStart(inst *serviceInstruction, sysd systemd.Systemd) Response {
+	// Refuse to start non-snap services
+	for _, service := range inst.Services {
+		if !strings.HasPrefix(service, "snap.") {
+			return InternalError("cannot start service %v", service)
```

Comment:
```suggestion
			return InternalError("cannot start non-snap service %v", service)
```

Reply:
Good idea.  I've made the same change to the error in `serviceStop`.